### PR TITLE
[WIP] Update docs and add new docs on ExtensionApp

### PR DESCRIPTION
**Pull request by Zsailer**
This PR refactors Jupyter Server's documentation pages. Most of this PR is removing old notebook documentation. My goal is to slim down the documentation to focus only on Jupyter Server. 

A major addition is that I've added a new page that explains the `ExtensionApp` merged in #48

### CODE REVIEW

**Comment by willingc** [comment]
Thanks for removing the unrelated docs. Was just going to ask you about that @Zsailer

File: docs/source/frontends.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
+Creating a Jupyter Server Frontend
+==================================
+
+Jupyter Server does not come with frontend out-of-the-box; instead, frontends are installed separately and loaded as server extensions. This page demonstrates the best way to write a Jupyter Server frontend from scatch (warning, this documentation is directed a developers).
```
Comment (willingc) [inline docs/source/frontends.rst]:
```suggestion
Jupyter Server does not come with a frontend out-of-the-box; instead, a frontend is installed separately and loaded as a server extension. This page demonstrates the best way to write a Jupyter Server frontend from scratch.

.. note::  This documentation is written for experienced developers.

```

File: docs/source/frontends.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+The ExtensionApp:
+
+    - can have traits.
+    - is configurable (from file or CLI)
+    - creates a namespace for the frontend's static files under ``/static/<extension_name>/``.
+    - loads itself as a server extension beside other Jupyter frontends and extensions.
+    - provides a command-line interface to launch the frontend extension directly (starting a server along the way).
+
+To create a new Jupyter frontend application, subclass the ExtensionApp like the example below:
```
Comment (willingc) [inline docs/source/frontends.rst]:
```suggestion
To create a new Jupyter frontend application, subclass the ``ExtensionApp`` like the example below:
```

File: docs/source/frontends.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            ...
+            # Change the jinja templating environment
+            self.settings.update({'myfrontend_jinja2_env': ...}) 
+
+The ``ExtensionApp`` uses the following methods and properties to connect your frontend to the Jupyter server. Overwrite these methods to add your custom settings, handlers and templates:
+
+* ``initialize_setting()``: adds custom settings to the Tornado Web Application.
+* ``initialize_handlers()``: appends handlers to the Tornado Web Application.
+* ``initialize_templates()``: initialize the templating engine (e.g. jinja2) for your frontend.
+* ``name``: the name of the extension
```
Comment (willingc) [inline docs/source/frontends.rst]: I would suggest separating the properties from the methods.

File: docs/source/frontends.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+To launch the application, simply call the ``ExtensionApp``'s ``launch_instance`` method.
+
+.. code-block:: python
+
+    main = MyFrontend.launch_instance
+    main()
+
+
+To make your frontend executable from anywher on your system, added this method as an entry-point oin your application's ``setup.py``
```
Comment (willingc) [inline docs/source/frontends.rst]:
```suggestion
To make your frontend executable from anywhere on your system, added this method as an entry-point in your application's ``setup.py``:
```

**Comment by Zsailer** [comment]
Thanks, @willingc!

**Comment by Zsailer** [comment]
Merging for tomorrow's meeting per @willingc's request 😎

**Comment by Zsailer** [comment]
I'll iterate on the docs more later.

**Comment by willingc** [comment]
Huge improvement. Thanks @Zsailer!

**Pull request merged**